cmd/create/oidcprovider/cmd.go

```diff
@@ -84,10 +84,21 @@ func modeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]str
 	return modes, cobra.ShellCompDirectiveDefault
 }
 
-func run(cmd *cobra.Command, _ []string) {
+func run(cmd *cobra.Command, argv []string) {
 	reporter := rprtr.CreateReporterOrExit()
 	logger := logging.CreateLoggerOrExit(reporter)
 
+	// Allow the command to be called programmatically
+	skipInteractive := false
+	if len(argv) == 2 && !cmd.Flag("cluster").Changed {
+		args.clusterKey = argv[0]
+		args.mode = argv[1]
+
+		if args.mode != "" {
+			skipInteractive = true
```

Comment:
It just occurred to me that if the user explicitly sets _both_ `--interactive` and `--mode`, we might want to actually display the interactive mode here. What do you think?

Reply:
having played around with this, it leads to repeating inputs, such as the `mode` and `permissions-boundary`, running through this flow we don't need any input from the user for operator roles or oidc provider except for confirmation on creating the resources. 

So going forward, do you think its best to : 
1. Leave as is, and override the explicit --interactive with the --mode param
2. Allow --interactive take priority and repeat inputs
3. Add explicit checks to inputs for both interactive and mode.

WDYT

Reply:
Skipped the permissions boundary prompt